DOC: update doc/readme.txt & remove devel/readme (#27631)
matplotlib/matplotlib: doc/README.txt

This is the top level build directory for the Matplotlib

> documentation.  All of the documentation is written using sphinx, a

Reviewer: ```suggestion
documentation.  All of the documentation is written using Sphinx, a
```

python documentation system built on top of ReST.  This directory contains the
following

Files
^^^^^

* index.rst - the top level include document (and landing page) for the Matplotlib docs

* conf.py - the sphinx configuration

* docutils.conf - htmnl output configuration

* Makefile and make.bat - entry points for building the docs

* matplotlibrc - rcParam configuration for docs

* missing-references.json - list of known missing/broken references

Content Folders
^^^^^^^^^^^^^^^

> * api - placeholders to automatically generate the api documentation

Reviewer: Maybe
```suggestion
* api - source files to automatically generate the api documentation
```

Reviewer: Works for me, though they are not templates in the same sense as `_templates`.

* devel - documentation for contributing to Matplotlib

* project - about Matplotlib, e.g. mission, code of conduct, licenses, history, etc.

* users - the user documentation, e.g., installation, plotting tutorials,
configuration tips, faq, explanations, etc.

* thirdpartypackages - redirect to <https://matplotlib.org/mpl-third-party/>

> Build Folders

Reviewer: ```suggestion
Build folders
```

^^^^^^^^^^^^^

> * _static - used by the sphinx build system

Reviewer: ```suggestion
* _static - supplementary non-content files; e.g. images, CSS, etc.
```

> * _templates - used by the sphinx build system

Reviewer: ```suggestion
* _templates - Sphinx page templates
```

* sphinxext - Sphinx extensions for the Matplotlib docs

Symlinks
--------

During the documentation build, sphinx-gallery creates symlinks from the source folders
in `/galleries` to target_folders in '/doc'; therefore ensure that you are editing the
real files rather than the symbolic links:

> 

Reviewer: I find the table format quite noisy. Would you consider a simple table or a bullet list? If you rather keep the fully outlined table, please separate header and content using = signs.
